python/dask_cudf/dask_cudf/backends.py

```diff
@@ -429,7 +429,9 @@ def union_categoricals_cudf(
 
 @_dask_cudf_nvtx_annotate
 def safe_hash(frame):
-    return cudf.Series(frame.hash_values(), index=frame.index)
+    return cudf.Series._from_data(
+        frame.hash_values()._data, index=frame._index
+    )
```

Comment:
Is this to avoid any "reindex with duplicates" issue?

Reply:
Yeah, some a dask pytest was failing:
```python
In [5]: s = cudf.Series([1, 2, 1, 2, 1, 4], index=[0, 0, 0, 0, 0, 0])

In [6]: s
Out[6]: 
0    1
0    2
0    1
0    2
0    1
0    4
dtype: int64

In [7]: s.hash_values()
Out[7]: 
0    4047427325
0    1683430573
0    4047427325
0    1683430573
0    4047427325
0    1851360991
dtype: uint32
```

Reply:
>a dask pytest was failing

I guess I'm curious *which* test was failing, since it's not immediately obvious to me that there is anything wrong with `hash_values` preserving the non-unique index.

Reply:
The pytest is: `python/dask_cudf/dask_cudf/tests/test_groupby.py::test_groupby_shuffle`


Here is the call stack:
```python
> /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/dask/dataframe/shuffle.py(939)shuffle_group()
-> ind = hash_object_dispatch(df[cols] if cols else df, index=False)
(Pdb) df
          a  b___count  b___sum  c___max
0     Alice          3     3072     1034
1       Bob          4     3900     1018
2   Charlie          6     6112     1054
3       Dan          2     1898     1001
4     Edith          1      920     1023
..      ...        ...      ...      ...
21   Victor          2     2044      983
22    Wendy          1      948      993
23   Xavier          3     2932     1053
24   Yvonne          3     2916     1008
25    Zelda          6     5925     1034

[199 rows x 4 columns]
(Pdb) df.sort_index()
         a  b___count  b___sum  c___max
0    Alice          3     3072     1034
0    Alice          3     2919     1080
0    Alice          3     2964     1014
0    Alice          3     2989     1030
0    Alice          2     1984     1020
..     ...        ...      ...      ...
24   Zelda          5     4872     1044
24  Yvonne          3     2916     1008
25   Zelda          3     2901     1033
25   Zelda          5     5079     1048
25   Zelda          6     5925     1034

[199 rows x 4 columns]
```

Reply:
Actually nope. `hash_values` already does the index handling for us. It's not necessary to re-assign/re-index again here. Dropped all unnecessary code.